matchzoo/datapack.py

```diff
@@ -53,18 +62,48 @@ def __init__(self,
         :param left: Store the content or features for id_left.
         :param right: Store the content or features for
             id_right.
-        :param context: Hyper-parameter fitted during
-            pre-processing stage.
         """
         self._relation = relation
         self._left = left
         self._right = right
-        self._context = context
+
+    @property
+    def has_label(self):
+        return 'label' in self._relation.columns
 
     def __len__(self) -> int:
         """Get numer of rows in the class:`DataPack` object."""
         return self._relation.shape[0]
 
+    @property
+    def frame(self):
+        return DataPackFrameView(self)
+
+    def unpack(self):
+        frame = self.frame[:]
+
+        columns = list(frame.columns)
+        if self.has_label:
+            columns.remove('label')
+            y = np.array(frame['label'])
+        else:
+            y = None
+
+        x = frame[columns].to_dict(orient='list')
+        for key, val in x.items():
+            x[key] = np.array(val)
+
+        return x, y
```

Comment:
machine learning community usually use X, y to denote training data and label, since X is actually a matrix, y is vector (notation from linear algebra).